pkg/virt-controller/watch/migration.go

```diff
@@ -402,36 +413,50 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 	} else if vmi == nil {
 		migrationCopy.Status.Phase = virtv1.MigrationFailed
 		c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedMigrationReason, "Migration failed because vmi does not exist.")
+		decPendingOrRunningMetric(migration, vmi, pod)
+		metrics.IncFailedMigrations(vmi, pod)
 		log.Log.Object(migration).Error("vmi does not exist")
 	} else if vmi.IsFinal() {
 		migrationCopy.Status.Phase = virtv1.MigrationFailed
 		c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedMigrationReason, "Migration failed vmi shutdown during migration.")
+		decPendingOrRunningMetric(migration, vmi, pod)
+		metrics.IncFailedMigrations(vmi, pod)
```

Comment:
Whenever migration fails, the phase of the migartion copy becomes `virtv1.MigrationFailed`:
`migrationCopy.Status.Phase = virtv1.MigrationFailed`

Therefore, maybe it's better to simply check whether the migration had failed at the end of the function instead of checking it here in every "if"